packages/common/src/utils/nft.utils.ts

```diff
@@ -173,4 +186,89 @@ export class NftUtils {
+   */
+  static async processNftEvent(
+    eventData: FullNodeTransaction,
+    stage: string,
+    network: unknown,
+    logger: Logger
+  ): Promise<boolean> {
+    // Early return if NFT auto review is disabled
+    if (!isNftAutoReviewEnabled()) {
+      logger.debug('NFT auto review is disabled. Skipping NFT handler invocation.');
+      return false;
+    }
+
+    // Early return if not a token creation transaction
+    if (eventData.version !== constants.CREATE_TOKEN_TX_VERSION) {
+      logger.debug(`Transaction version ${eventData.version} is not a token creation transaction (${constants.CREATE_TOKEN_TX_VERSION}). Skipping NFT handler invocation.`);
+      return false;
+    }
+
+    try {
+      // Transform the data to a format compatible with shouldInvokeNftHandlerForTx
+      const transformedTx = NftUtils.transformFullNodeTxForNftDetection(eventData);
+
+      // Check if we should invoke the NFT handler for this transaction
+      if (NftUtils.shouldInvokeNftHandlerForTx(transformedTx, network, logger)) {
+        // Get the transaction hash
+        const txId = eventData.hash;
+
+        // Invoke the lambda function
+        await NftUtils.invokeNftHandlerLambda(txId, stage, logger);
+        return true;
+      }
+
+      return false;
+    } catch (error) {
+      logger.error(`Error processing NFT event: ${error}`);
+      return false;
+    }
+  }
+
+  /**
+   * Transform transaction data from the full node to a format compatible with the NFT detection logic.
+   */
+  static transformFullNodeTxForNftDetection(fullNodeData: FullNodeTransaction): HistoryTransaction {
+    // Create a new object with the required properties
+    const transformedTx: any = {
+      hash: fullNodeData.hash,
+      tx_id: fullNodeData.hash, // Add tx_id for compatibility
+      version: fullNodeData.version,
+      tokens: fullNodeData.tokens,
+      token_name: fullNodeData.token_name,
+      token_symbol: fullNodeData.token_symbol,
+      inputs: fullNodeData.inputs.map((input: FullNodeInput) => {
+        const tokenIndex = (input.spent_output.token_data & constants.TOKEN_INDEX_MASK) - 1;
+
+        return {
+          tx_id: input.tx_id,
+          index: input.index,
+          token: tokenIndex < 0 ? constants.HATHOR_TOKEN_CONFIG.uid : fullNodeData.tokens[tokenIndex],
+          token_data: input.spent_output.token_data,
+          value: input.spent_output.value,
+          script: input.spent_output.script,
+          decoded: input.spent_output.decoded || {},
+        };
+      }),
+      outputs: fullNodeData.outputs.map((output: FullNodeOutput) => {
+        const tokenIndex = (output.token_data & constants.TOKEN_INDEX_MASK) - 1;
+
+        return {
+          value: output.value,
+          token_data: output.token_data,
+          script: output.script,
+          token: tokenIndex < 0 ? constants.HATHOR_TOKEN_CONFIG.uid : fullNodeData.tokens[tokenIndex],
+          decoded: output.decoded || {},
+          spent_by: null,
```

Comment:
question: Is there no possibility that an event may arrive with a spent token creation tx? Not even an update?

Reply:
No, the fullnode sends events in the order they're processed

Reply:
I think it's irrelevant in this context, we're just transforming the tx here so it's processed properly by the wallet-lib's method